kura/org.eclipse.kura.asset.provider/src/main/java/org/eclipse/kura/asset/provider/BaseAsset.java

```diff
@@ -542,41 +507,45 @@ private AssetRecord createAssetRecordFromDriverRecord(DriverRecord driverRecord)
 
     /** {@inheritDoc} */
     @Override
-    public List<AssetRecord> read(final List<Long> channelIds) throws KuraException {
+    public List<AssetRecord> read(final Set<String> channelNames) throws KuraException {
         requireNonNull(this.driver, message.driverNonNull());
```

Comment:
If I'm not wrong the exception was not thrown even before this PR.
In my opinion there is no need to throw an exception in this case, returning an empty list is fine. I would prefer removing the IllegalArgumentException from the javadoc instead.